toolchain/parse/typed_nodes_test.cpp

```diff
@@ -284,5 +284,84 @@ TEST_F(TypedNodeTest, CategoryMatches) {
 #include "toolchain/parse/node_kind.def"
 }
 
+TEST_F(TypedNodeTest, Choice) {
```

Comment:
We don't generally expect unit tests for each kind of typed node. We check that the file tests (eg, testdata/choice/basic.carbon) can be successfully extracted as typed nodes as part of parse tree verification, so there is test coverage for extraction of these new typed nodes even without this added test.